app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,15 @@
+class CategoriesController < ApplicationController
+  before_action :require_admin
```

Comment:
@alischlereth @iandouglas The way I currently have this set up, this controller only handles destroying a category, which only an admin can do. Do i need to nest this in an admin/categories controller with namespacing? Or since I don't care about the path (you are on gifs index when you click to delete, and remain on that page)  is it ok to structure this way?

Reply:
if you have a before_action that requires an admin user (I'm presuming you have that defined in your ApplicationController then?) then this should work out okay. Be sure you set up a 404 redirect if the user isn't an admin, though.